app/controllers/RestaurantController.java

```diff
@@ -115,7 +116,13 @@ public Result getNearbyRestaurants(Float latitude, Float longitude) {
 	 */
 	@Transactional(readOnly = true)
 	public Result getPopularRestaurants() {
-		return wrapForPublic(() -> this.service.getPopularRestaurants());
+	        
+	      
+	        User user = this.cache.get(session("uid"));
+	      //  String id = user.getId().toString();
+		return wrapForPublic(() -> this.service.getPopularRestaurants(user));
```

Comment:
Should be wrap for user, because this can cause a issues if called when no user is logged in

Reply:
U situaciji kada stavim wrapForUser, imam problem kada nije niko ulogovan.